airflow/models/taskinstancehistory.py

```diff
@@ -0,0 +1,128 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from sqlalchemy import (
+    Column,
+    ForeignKeyConstraint,
+    Integer,
+    PrimaryKeyConstraint,
+    func,
+    select,
+    text,
+)
+
+from airflow.models.base import Base, StringID
+from airflow.models.taskinstance import TaskInstance
+from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.utils.state import State, TaskInstanceState
+
+if TYPE_CHECKING:
+    from airflow.serialization.pydantic.taskinstance import TaskInstancePydantic
+
+
+class TaskInstanceHistory(Base):
+    """
+    Store old tries of TaskInstances.
+
+    :meta private:
+    """
+
+    __tablename__ = "task_instance_history"
```

Comment:
Let's give this new table a single pk column - integer or uuid? Everything else that doesn't have it we are regretting right now

Reply:
Done